modules/apps/saved-content/saved-content-web/src/main/java/com/liferay/saved/content/web/internal/portlet/action/EditSavedContentMVCActionCommand.java

```diff
@@ -0,0 +1,107 @@
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.language.Language;
+import com.liferay.portal.kernel.log.Log;
+import com.liferay.portal.kernel.log.LogFactoryUtil;
+import com.liferay.portal.kernel.portlet.JSONPortletResponseUtil;
+import com.liferay.portal.kernel.portlet.bridges.mvc.BaseMVCActionCommand;
+import com.liferay.portal.kernel.portlet.bridges.mvc.MVCActionCommand;
+import com.liferay.portal.kernel.service.ServiceContextFactory;
+import com.liferay.portal.kernel.theme.ThemeDisplay;
+import com.liferay.portal.kernel.util.ParamUtil;
+import com.liferay.portal.kernel.util.Portal;
+import com.liferay.portal.kernel.util.WebKeys;
+import com.liferay.saved.content.constants.MySavedContentPortletKeys;
+import com.liferay.saved.content.model.SavedContentEntry;
+import com.liferay.saved.content.service.SavedContentEntryService;
+
+import javax.portlet.ActionRequest;
+import javax.portlet.ActionResponse;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Alicia García
+ */
+@Component(
+	property = {
+		"javax.portlet.name=" + MySavedContentPortletKeys.MY_SAVED_CONTENT,
+		"mvc.command.name=/saved_content/edit_saved_content"
+	},
+	service = MVCActionCommand.class
+)
+public class EditSavedContentMVCActionCommand extends BaseMVCActionCommand {
+
+	@Override
+	protected void doProcessAction(
+			ActionRequest actionRequest, ActionResponse actionResponse)
+		throws Exception {
+
+		JSONPortletResponseUtil.writeJSON(
+			actionRequest, actionResponse,
+			_updateSavedContentEntry(actionRequest));
+	}
+
+	private JSONObject _updateSavedContentEntry(ActionRequest actionRequest) {
+		String className = ParamUtil.getString(actionRequest, "className");
+		long classPK = ParamUtil.getLong(actionRequest, "classPK");
+
+		ThemeDisplay themeDisplay = (ThemeDisplay)actionRequest.getAttribute(
+			WebKeys.THEME_DISPLAY);
+
+		long userId = themeDisplay.getUserId();
+
+		try {
+			SavedContentEntry savedContentEntry =
+				_savedContentEntryService.fetchSavedContentEntry(
+					themeDisplay.getCompanyId(), userId, className, classPK);
+
+			if (savedContentEntry == null) {
+				_savedContentEntryService.addSavedContentEntry(
+					themeDisplay.getScopeGroupId(), userId, className, classPK,
+					ServiceContextFactory.getInstance(
+						SavedContentEntry.class.getName(), actionRequest));
+
+				return JSONUtil.put("saved", Boolean.TRUE);
+			}
+
+			_savedContentEntryService.deleteSavedContentEntry(
+				savedContentEntry);
+
+			return JSONUtil.put("saved", Boolean.FALSE);
+		}
+		catch (PortalException portalException) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(portalException);
+			}
+
+			return JSONUtil.put(
+				"errorMessage",
+				_language.get(
+					_portal.getHttpServletRequest(actionRequest),
+					"an-unexpected-error-occurred"));
+		}
```

Comment:
If you are going to do more changes, I would change it as written before:

> By the way, since we already have the ThemeDisplay I think is better to directly use _language.get(themeDisplay.getLocale(), ...) because the actual call ends getting the themeDisplay anyway.